### PR TITLE
[Jetpack Remote Install] Install view

### PR DESCRIPTION
**Pull request by danielebogo**
Refs. #10277 

jetpack-states

## Jetpack view
This PR is about the Jetpack Install view, used to display the different states during the Jetpack installation.

There are _4_ states which are represented by the `JetpackRemoteInstallViewState` enum:
- install
- installing
- success
- failure

### Install
This is the default state, the initial state when the controller is presented.

### Installation
This is the state indicating that the service is installing Jetpack

### Success
This is the state to indicate the success result of the service operation. From here the user will open the next controller to complete the Jetpack setup.

### Failure
This is an associated type because the _case_ accept a `JetpackInstallError`. It indicates an error occurred while the service was working. From here the user can open the customer support or retry.

## Small Design
`JetpackRemoteInstallViewController` is the main view controller. It communicates with the view model and it's in charge to push some other _section_. Its child `JetpackRemoteInstallView` is a view controller used as passive view. `JetpackRemoteInstallViewModel` is the skeleton of the view model which will include the service (or a store using Flux) in the next PR.

## Errors
`JetpackInstallError+Blocking` is an extension used to identify if an error is a _blocking error_. If it is then we fallback the action to the original web view.

## Warnings
There are 4 `#warning` I added. This are a reminder to remove some method and some UI element, integrated only to help the PR reviewer with the test. Everything in this PR is behind feature flag, so no worries 😊

**To Test**
• Add a self-hosted site with no Jetpack installed
• Open _Stats_ from your site dashboard
• Click on _Install Jetpack_
• After the view controller is presented, use the segmented control to switch between the different states. The last case simulate when the error is *blocking* and will dismiss the current screen and open a webview. The Customer support button will open the support view controller

### CODE REVIEW

**Comment by danielebogo** [comment]
@SylvesterWilmott  these are the screenshots:
Simulator Screen Shot - iPhone X - 2019-04-04 at 15 25 08

Simulator Screen Shot - iPhone X - 2019-04-04 at 15 25 10

Simulator Screen Shot - iPhone X - 2019-04-04 at 15 25 13

Simulator Screen Shot - iPhone X - 2019-04-04 at 15 25 16

File: WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallView.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            supportButton.isHidden = false
+        default:
+            supportButton.isHidden = true
+        }
+    }
+}
+
+private extension JetpackRemoteInstallView {
+    func setupUI() {
+        view.backgroundColor = WPStyleGuide.itsEverywhereGrey()
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallView.swift]: This one too.

File: WordPress/Classes/ViewRelated/Jetpack/Install/JetpackRemoteInstallViewController.swift [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+private extension JetpackRemoteInstallViewController {
+    func setupNavigationBar() {
+        title = NSLocalizedString("Jetpack", comment: "Title for the Jetpack Installation")
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
+                                                           target: self,
+                                                           action: #selector(cancel))
+    }
+
+    func setupUI() {
+        view.backgroundColor = WPStyleGuide.itsEverywhereGrey()
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply WordPress/Classes/ViewRelated/Jetpack/Install/JetpackRemoteInstallViewController.swift]: Oh yes absolutely! I got it from the first view controller used!

**Comment by SylvesterWilmott** [comment]
Looks good @danielebogo 

I think there's a small issue with the spacing, to follow the specs from our empty states:

- The space between title and subtitle should be 10pt
- The space between subtitle and button should be 20pt

File: WordPress/Classes/ViewRelated/Jetpack/Install/Data/JetpackRemoteInstallViewState.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    var title: String {
+        switch self {
+        case .install:
+            return NSLocalizedString("Install Jetpack", comment: "The default Jetpack view title")
+        case .installing:
+            return NSLocalizedString("Installing Jetpack", comment: "The Jetpack view title used while the is installing")
+        case .failure:
+            return NSLocalizedString("There was a problem", comment: "The Jetpack view title used when an error occurred")
+        case .success:
+            return NSLocalizedString("Jetpack Installed", comment: "The Jetpack view title for the success state")
```
[the remark this answers is not in the document]
Comment (SylvesterWilmott) [reply WordPress/Classes/ViewRelated/Jetpack/Install/Data/JetpackRemoteInstallViewState.swift]: Ah yes, well spotted @aerych 

Lowercase "i" for sure

**Comment by danielebogo** [comment]
@SylvesterWilmott 

Simulator Screen Shot - iPhone X - 2019-04-08 at 10 57 34

**Comment by SylvesterWilmott** [comment]
Looks good @danielebogo , thanks!

**Pull request merged**